### PR TITLE
Fix role name config

### LINKED ISSUE #1444: cloud_RoleName blank in app insights portal under 3.0.1 (worked in 3.0.1-BETA.2)
### Expected behavior
Setting the role.name in applicationinsights.json should set the `cloud_RoleName` in application insights portal.  This was working in 3.0.1-BETA.2, but stopped working when we upgraded to 3.0.1.  It looks like it was caused by a bug introduced in the fix for https://github.com/microsoft/ApplicationInsights-Java/issues/1425, specifically in https://github.com/microsoft/ApplicationInsights-Java/blob/master/agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java#L196-L200 line 198 should set the instance rather than the name:
```
            config.role.instance = getEnv(WEBSITE_INSTANCE_ID);
```

### Actual behavior
The `cloud_RoleName` field is blank in application insights

### To Reproduce
In applicationinsights.json set the role name:
```
  "role": {
    "name": "my-application"
  },
```

Start the application, make a request to it and wait for the request to appear in app insights:
```
requests
| where cloud_RoleInstance == "my-pc"
| project timestamp, cloud_RoleName
```

The `cloud_RoleName` is then blank under 3.0.1.

Switching to the 3.0.1-BETA.2 agent and running the same test then correctly displays the `cloud_RoleName`.

Specifying an environment variable of `APPLICATIONINSIGHTS_ROLE_NAME` as documented in https://github.com/microsoft/ApplicationInsights-Java/issues/1425 also then correctly displays the `cloud_RoleName`.

#### Sample Application
Project to replicate the issue is https://github.com/ministryofjustice/prison-estate.git. Need Java 11. Replace with your key for app insights.

Demonstrating working under agent 3.0.1-BETA.2.
```
git checkout pgp-DT-1482-insights-v3
./gradlew clean assemble && cp applicationinsights.json build/libs
APPLICATIONINSIGHTS_CONNECTION_STRING=InstrumentationKey=<your instrumentation key> java -javaagent:build/libs/applicationinsights-agent-3.0.1-BETA.2.jar -Dspring.profiles.active=dev -jar build/libs/prison-estate-*.jar
curl http://localhost:8080/prisons/id/MDI
```

Similarly not working under 3.0.1:
```
git checkout pgp-DT-1442-insights-v3.0.1
./gradlew clean assemble && cp applicationinsights.json build/libs
APPLICATIONINSIGHTS_CONNECTION_STRING=InstrumentationKey=<your instrumentation key> java -javaagent:build/libs/applicationinsights-agent-3.0.1.jar -Dspring.profiles.active=dev -jar build/libs/prison-estate-*.jar
curl http://localhost:8080/prisons/id/MDI
```

In app insights
```
requests
| project cloud_RoleName
```
should then show the role name, but it is only populated for the 3.0.1-BETA.2 requests.

### System information
Please provide the following information:
- SDK Version: 2.6.2
- Java Agent Version: 3.0.1
- OS type and version: Debian 14
- Application Server type and version (if applicable): Spring Boot 2.4.1
- Using spring-boot? yes

### PR DESCRIPTION
**Pull request by trask**
Resolves #1444

### CODE REVIEW

**Review by kryalama — APPROVED** [review]

File: agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public static void overlayEnvVars(Configuration config) throws IOException {
 
         if (isTrimEmpty(config.role.instance)) {
             // only use WEBSITE_INSTANCE_ID as a fallback
-            config.role.name = getEnv(WEBSITE_INSTANCE_ID);
+            config.role.instance = getEnv(WEBSITE_INSTANCE_ID);
```
Comment (heyams) [inline agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java:198]: can we add a test for this change? a config with role name?

Comment (trask) [reply agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java:198]: see new tests in this PR

File: agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java [hunk 2]
```diff
@@ -176,6 +182,68 @@ public void shouldOverrideConnectionString() throws IOException {
...
 
+    @Test
+    public void shouldOverrideRoleName() throws IOException {
+        envVars.set("APPLICATIONINSIGHTS_ROLE_NAME", "role name from env");
+        envVars.set("WEBSITE_SITE_NAME", "role name from website env");
+
+        Configuration configuration = loadConfiguration();
+        ConfigurationBuilder.overlayEnvVars(configuration);
+
+        assertEquals("role name from env", configuration.role.name);
```
Comment (heyams) [inline agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: should it be configuration.role.instance?

Comment (trask) [reply agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: see test below for `configuration.role.instance` test

Comment (heyams) [reply agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: changed 
```{
  "connectionString": "InstrumentationKey=...",
  "role": {
    "**name**": "my cloud role name"
  }
}
```

do we need to update the microsoft doc to reflect this change?

to 

```{
  "connectionString": "InstrumentationKey=...",
  "role": {
    "**instance**": "my cloud role name"
  }
}
```

Comment (trask) [reply agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: role `name` and role `instance` are two different fields

Comment (heyams) [reply agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: i was confused.  instance field is not exposed in the json config file?

Comment (trask) [reply agent/agent-bootstrap/src/test/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationTest.java:193]: it is exposed in json config file: https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#cloud-role-instance

**Review by heyams — APPROVED** [review]

**Pull request merged**